go/app/main.go

```diff
@@ -28,14 +46,71 @@ func root(c echo.Context) error {
 func addItem(c echo.Context) error {
 	// Get form data
 	name := c.FormValue("name")
+	category := c.FormValue("category")
+	image, error := c.FormFile("image")
+	if error != nil {
+		return c.JSON(http.StatusBadRequest, error)
+	}
+
 	c.Logger().Infof("Receive item: %s", name)
+	c.Logger().Infof("Receive category: %s", category)
+	c.Logger().Infof("Receive image: %s", image.Filename)
+
+	updateJson(name, category, image)
+	saveImageToLocal(image)
 
 	message := fmt.Sprintf("item received: %s", name)
 	res := Response{Message: message}
 
 	return c.JSON(http.StatusOK, res)
 }
 
+func getItems(c echo.Context) error {
+	jsonFile, err := os.Open(ItemFile)
+	if err != nil {
+		fmt.Println("Cannot open the json file", err)
+		return err
+	}
+	defer jsonFile.Close()
+
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		fmt.Println("Cannot read data", err)
+		return err
+	}
+
+	var items Items
+
+	json.Unmarshal(jsonData, &items)
+
+	return c.JSON(http.StatusOK, items)
+}
+
+func getItemsByID(c echo.Context) error {
+	id := c.Param("itemID")
+	jsonFile, err := os.Open(ItemFile)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, err)
+	}
+
+	defer jsonFile.Close()
+
+	jsonData, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, err)
+	}
+
+	var items Items
+	json.Unmarshal(jsonData, &items)
```

Comment:
Since lines 90-104 are repeated in `getItems`, `getItemsByID` and `updateJson`, maybe it would be better to extract it into a separate function! e.g `fetchItems()`. This can then be used in the above mentioned functions to reduce duplicate code!